src/WPF/WPF.Viewer/Samples/Data/EditFeaturesUsingFeatureForms/EditFeaturesUsingFeatureForms.xaml.cs

```diff
@@ -0,0 +1,132 @@
+        tags: new[] { "edits", "feature", "featureforms", "form", "toolkit" })]
+    public partial class EditFeaturesUsingFeatureForms : UserControl
+    {
+        private FeatureForm _featureForm;
+
+        public EditFeaturesUsingFeatureForms()
+        {
+            InitializeComponent();
+            _ = Initialize();
+        }
+
+        private async Task Initialize()
+        {
+            // Create the ArcGIS Online portal.
+            ArcGISPortal portal = await ArcGISPortal.CreateAsync();
+
+            // Get the Naperville water web map item using its ID.
+            PortalItem webmapItem = await PortalItem.CreateAsync(portal, "516e4d6aeb4c495c87c41e11274c767f");
+
+            // Create a map from the web map item.
+            Map onlineMap = new Map(webmapItem);
+
+            // Display the map in the MapView.
+            MyMapView.Map = onlineMap;
+        }
+
+        private async void MyMapView_GeoViewTapped(object sender, GeoViewInputEventArgs e)
+        {
+            try
+            {
+                // Perform identify operation to get the feature.
+                var identifyResult = await MyMapView.IdentifyLayersAsync(e.Position, 12, false);
+                var feature = identifyResult.SelectMany(result => result.GeoElements).OfType<ArcGISFeature>().FirstOrDefault();
+
+                if (feature != null)
+                {
+                    // Create a feature form.
+                    _featureForm = new FeatureForm(feature);
+
+                    // Add the feature form view to the scroll viewer.
+                    FeatureFormViewPanel.FeatureForm = _featureForm;
+
+                    // Show the feature form panel.
+                    FeatureFormPanel.Visibility = Visibility.Visible;
+                }
+            }
+            catch (Exception ex)
+            {
+                MessageBox.Show($"Error: {ex.Message}", "Error", MessageBoxButton.OK, MessageBoxImage.Error);
+            }
+        }
+
+        private async void SaveButton_Click(object sender, RoutedEventArgs e)
+        {
+            try
+            {
+                // Check if there are validation errors.
+                if (_featureForm.ValidationErrors.Any())
+                {
+                    var errors = _featureForm.ValidationErrors;
+                    var errorMessages = errors.SelectMany(kvp => kvp.Value.Select(ex => $"{kvp.Key}: {ex.Message}"));
+                    string errorMessage = string.Join("\n", errorMessages);
+                    throw new Exception($"Validation errors exist.\n{errorMessage}");
+                }
+
+                // Finish editing.
+                await _featureForm.FinishEditingAsync();
+
+                // Get the service feature table.
+                var serviceFeatureTable = (ServiceFeatureTable)_featureForm.Feature.FeatureTable;
```

Comment:
This code from here and down is only needed if it's a service feature table. Local tables won't need any of this. Could we make that more clearer? Perhaps something like.
if (_featureForm.Feature.FeatureTable is ServiceFeatureTable serviceFeatureTable)
{
    // if this is a service feature table, make sure the changes are also pushed back up to the server
}